ts/screens/profile/components/services/ServicesContactComponent.tsx

```diff
@@ -15,73 +16,82 @@ import ItemSeparatorComponent from "../../../../components/ItemSeparatorComponen
 import { H5 } from "../../../../components/core/typography/H5";
 import TouchableDefaultOpacity from "../../../../components/TouchableDefaultOpacity";
 import I18n from "../../../../i18n";
+import { profileSelector } from "../../../../store/reducers/profile";
+import { ServicesPreferencesModeEnum } from "../../../../../definitions/backend/ServicesPreferencesMode";
 
 type Props = {
-  onSelectOption: (optionKey: string) => void;
+  onSelectMode: (mode: ServicesPreferencesModeEnum) => void;
   hasAlreadyOnboarded?: true;
 } & ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>;
 
 type ContactOption = {
   title: string;
-  key: string;
+  mode: ServicesPreferencesModeEnum;
   description1: string;
   description2?: string;
 };
 
 const options: ReadonlyArray<ContactOption> = [
   {
     title: I18n.t("services.optIn.preferences.quickConfig.title"),
-    key: "quick",
+    mode: ServicesPreferencesModeEnum.AUTO,
     description1: I18n.t("services.optIn.preferences.quickConfig.body.text1"),
     description2: I18n.t("services.optIn.preferences.quickConfig.body.text2")
   },
   {
     title: I18n.t("services.optIn.preferences.manualConfig.title"),
-    key: "manual",
+    mode: ServicesPreferencesModeEnum.MANUAL,
     description1: I18n.t("services.optIn.preferences.manualConfig.body.text1")
   }
 ];
 
 const ServicesContactComponent = (props: Props): React.ReactElement => {
-  const [selected, setSelected] = useState<string | undefined>();
-
-  const renderListItem = ({ item }: ListRenderItemInfo<ContactOption>) => (
-    <>
-      <TouchableDefaultOpacity
-        style={[
-          IOStyles.row,
-          {
-            justifyContent: "space-between"
+  const renderListItem = ({ item }: ListRenderItemInfo<ContactOption>) => {
```

Comment:
Given the size it has reached, `renderListItem` may well become a component by itself. This will incourage us to add at least one snapshot test to ensure its shape doesn't change inadvertently. 

A further interaction might be to add one test to ServicesContactComponent which is now uncovered: one happy path to ensure `onSelectMode` is called on click seems like a good candidate to me.